plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/persistence/LibraryClasspathContainerSerializer.java

```
@@ -0,0 +1,109 @@
+package com.google.cloud.tools.eclipse.appengine.libraries.persistence;
+
+import com.google.cloud.tools.eclipse.appengine.libraries.LibraryClasspathContainer;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.logging.Logger;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.jdt.core.IJavaProject;
+import org.osgi.framework.FrameworkUtil;
+
+public class LibraryClasspathContainerSerializer {
+
+  private static final Logger logger = Logger.getLogger(LibraryClasspathContainerSerializer.class.getName());
+
+  public interface LibraryContainerStateLocationProvider {
+    IPath getContainerStateFile(IJavaProject javaProject, IPath containerPath, boolean create) throws CoreException;
+  }
+
+  private LibraryContainerStateLocationProvider stateLocationProvider;
+  
+  public LibraryClasspathContainerSerializer() {
+    this(new DefaultStateLocationProvider());
+  }
+  
+  @VisibleForTesting
+  public LibraryClasspathContainerSerializer(LibraryContainerStateLocationProvider stateLocationProvider) {
+    this.stateLocationProvider = stateLocationProvider;
+  }
+
+  public void saveContainer(IJavaProject javaProject, LibraryClasspathContainer container) throws IOException,
+                                                                                                  CoreException {
+    File stateFile = getContainerStateFile(javaProject, container.getPath(), true);
+    if (stateFile == null) {
+      logger.warning("Container state file cannot be created, save failed");
+      return;
+    }
+    try (PrintWriter outputStream = new PrintWriter(stateFile)) {
```

don't use PrintWriter because it swallows exceptions and may not properly internationalize. Try OutputStreamWriter instead